### PR TITLE
[EPO-3029] Colorize the plot per multiple datasets

### PR DESCRIPTION
**Pull request by georgegomez144**

### CODE REVIEW

File: src/components/charts/sizeDistancePlotter/LegendMultiple.jsx [hunk 1]
```diff
@@ -7,21 +7,22 @@ import styles from './size-distance-plotter.module.scss';
 
 class LegendMultiple extends React.PureComponent {
   render() {
-    const { yValueAccessor, numOfSets } = this.props;
+    const { yValueAccessor, numOfSets, data } = this.props;
```
Comment (blnkt) [inline src/components/charts/sizeDistancePlotter/LegendMultiple.jsx:10]: numOfSets never gets used: delete

File: src/components/charts/sizeDistancePlotter/LegendMultiple.jsx [hunk 2]
```diff
@@ -7,21 +7,22 @@ import styles from './size-distance-plotter.module.scss';
...
     const useFluxColor = yValueAccessor === 'color';
 
     return (
       <Card className={styles.legend}>
-        {numOfSets &&
-          Array.from(Array(numOfSets)).map((set, i) => {
+        {data &&
+          numOfSets &&
```
Comment (blnkt) [inline src/components/charts/sizeDistancePlotter/LegendMultiple.jsx:16]: numOfSets never gets used: delete

File: src/components/charts/sizeDistancePlotter/LegendMultiple.jsx [hunk 3]
```diff
@@ -50,6 +51,7 @@ class LegendMultiple extends React.PureComponent {
 LegendMultiple.propTypes = {
   yValueAccessor: PropTypes.string,
   numOfSets: PropTypes.number,
```
Comment (blnkt) [inline src/components/charts/sizeDistancePlotter/LegendMultiple.jsx:53]: numOfSets never gets used: delete

File: src/components/charts/sizeDistancePlotter/index.jsx [hunk 4]
```diff
@@ -312,6 +312,7 @@ class SizeDistancePlotter extends React.Component {
           <LegendMultiple
             yValueAccessor={yValueAccessor}
             numOfSets={data.length}
```
Comment (blnkt) [inline src/components/charts/sizeDistancePlotter/index.jsx:314]: numOfSets never gets used: delete

File: src/components/charts/sizeDistancePlotter/LegendMultiple.jsx [hunk 5]
```diff
@@ -7,21 +7,22 @@ import styles from './size-distance-plotter.module.scss';
...
             const isDefaultSet = i === 0;
             const setId = `set-${i}`;
             const pointClasses = classnames(
               styles.legendPoint,
               setId,
               styles.groupPoint,
+              styles[`colorSet${i}`],
               {
-                [styles[`colorSet${i}`]]: i > 0,
                 [styles.defaultPoint]: isDefaultSet && !useFluxColor,
```
Comment (blnkt) [inline src/components/charts/sizeDistancePlotter/LegendMultiple.jsx:26]: this was only applicable to your color/brightness vs distance plots: delete

File: src/components/charts/sizeDistancePlotter/LegendMultiple.jsx [hunk 6]
```diff
@@ -7,21 +7,22 @@ import styles from './size-distance-plotter.module.scss';
 
 class LegendMultiple extends React.PureComponent {
   render() {
-    const { yValueAccessor, numOfSets } = this.props;
+    const { yValueAccessor, numOfSets, data } = this.props;
     const useFluxColor = yValueAccessor === 'color';
```
Comment (blnkt) [inline src/components/charts/sizeDistancePlotter/LegendMultiple.jsx:11]: this was only applicable to your color/brightness vs distance plots: delete

File: src/components/charts/sizeDistancePlotter/LegendMultiple.jsx [hunk 7]
```diff
@@ -7,21 +7,22 @@ import styles from './size-distance-plotter.module.scss';
...
 
     return (
       <Card className={styles.legend}>
-        {numOfSets &&
-          Array.from(Array(numOfSets)).map((set, i) => {
+        {data &&
+          numOfSets &&
+          data.map((datum, i) => {
             const isDefaultSet = i === 0;
```
Comment (blnkt) [inline src/components/charts/sizeDistancePlotter/LegendMultiple.jsx:18]: this was only applicable to your color/brightness vs distance plots: delete

File: src/components/charts/sizeDistancePlotter/LegendMultiple.jsx [hunk 8]
```diff
@@ -7,21 +7,22 @@ import styles from './size-distance-plotter.module.scss';
...
             const setId = `set-${i}`;
             const pointClasses = classnames(
               styles.legendPoint,
               setId,
               styles.groupPoint,
+              styles[`colorSet${i}`],
```
Comment (blnkt) [inline src/components/charts/sizeDistancePlotter/LegendMultiple.jsx:24]: I think we probably only want the colorful colors used so let's do `i + 1` to skip over the grey

File: src/components/charts/sizeDistancePlotter/LegendMultiple.jsx [hunk 9]
```diff
@@ -37,7 +38,7 @@ class LegendMultiple extends React.PureComponent {
                   }
                 ></span>
                 <span>{`${
-                  isDefaultSet ? 'Rubin Observatory Data' : 'Your Data'
+                  isDefaultSet ? 'Rubin Observatory Data' : datum.group
```
Comment (blnkt) [inline src/components/charts/sizeDistancePlotter/LegendMultiple.jsx:41]: after getting rid of `isDefaultSet` (see other comments) this probably becomes `datum.group || 'Rubin Observatory Data'`

File: src/data/pages/TEST-size-distance-plotter.json [hunk 10]
```diff
@@ -24,6 +24,7 @@
       },
```
Comment (blnkt) [inline src/data/pages/TEST-size-distance-plotter.json:24]: delete this test page and reimplement this plot on `http://localhost:8000/object-magnitude-solar-system/`.  Also, while you're updating that page, please fix this typo in the last question on that page: "...large object throught the solar system?" should be "...large objects throughout the solar system?"

File: src/components/charts/sizeDistancePlotter/index.jsx [hunk 11]
```diff
@@ -382,7 +383,7 @@ class SizeDistancePlotter extends React.Component {
                   return (
                     <Points
                       key={setId}
-                      data={set}
+                      data={set.data || set}
```
Comment (blnkt) [inline src/components/charts/sizeDistancePlotter/index.jsx:386]: I think probably this or is superfluous at this time.  We only have one json file to send to this plot and we're in charge of how it's structured so, for the time being, it's always `set.data`, right?  Or am I thinking about this wrong?

Comment (georgegomez144) [reply src/components/charts/sizeDistancePlotter/index.jsx:386]: I think this check, in this case, is certainly not need. I'm sure we'll remember to use this very basic method in the future if this ever requires the change.

**Pull request closed without merging**